step-release-vis/src/app/components/environment/environment_test.ts

```diff
@@ -410,15 +413,52 @@ describe('EnvironmentComponent', () => {
 
   describe('time range drag', () => {
     it('should update time range', () => {
-      // TODO(#277): implement test
+      const svg = fixture.debugElement.query(
+        By.css(`#${component.environment.name}-svg`)
+      );
+      spyOn(component.startTimestampChange, 'emit');
+      spyOn(component.endTimestampChange, 'emit');
+      svg.triggerEventHandler('mousedown', {pageX: 100, pageY: 100});
+      fixture.detectChanges();
+      expect(component.draggedEnvName.name).toEqual(component.environment.name);
+      svg.triggerEventHandler('mousemove', {pageX: 200, pageY: 200});
+      svg.triggerEventHandler('mouseup', {pageX: 200, pageY: 200});
+      fixture.detectChanges();
+      expect(component.startTimestampChange.emit).toHaveBeenCalled();
+      expect(component.endTimestampChange.emit).toHaveBeenCalled();
     });
 
     it('should not update, if mouse goes outside env and is released', () => {
-      // TODO(#277): implement test
+      const svg = fixture.debugElement.query(
+        By.css(`#${component.environment.name}-svg`)
+      );
+      spyOn(component.startTimestampChange, 'emit');
+      spyOn(component.endTimestampChange, 'emit');
+
+      svg.triggerEventHandler('mousedown', {pageX: 100, pageY: 100});
+      svg.triggerEventHandler('mousemove', {pageX: 200, pageY: 200});
+      component.draggedEnvName.name = 'other_env';
+
```

Comment:
I do not see mouseup here, is it the test, the test description or implementation which is wrong? :D

Reply:
The mouseup is supposed to happen in a sibling environment, can't do that explicitly in this test. I'll update the description.